src/DomainHost.sol

```diff
@@ -306,17 +306,45 @@ abstract contract DomainHost {
         emit Deficit(wad);
     }
 
+    /// @notice Governance is accruing surplus
+    /// @param _grain It should be the amount of grain necessary to cover debt in the guest domain [WAD]
+    function accrue(uint256 _grain) external auth returns (uint256 _wad) {
+        require(vat.live() == 1, "DomainHost/vat-not-live");
+        uint256 _dsin = dsin;
+        uint256 _ddai = ddai;
+        require(_ddai > _dsin, "DomainHost/no-surplus");
+
+        unchecked { _wad = _ddai - _dsin; }
+        dai.transferFrom(address(escrow), address(this), _wad);
```

Comment:
I think there are several ways to solve this, but the main problem is we can not trust with a permissionless method as a malicious domain can actually inform more surplus than actually exists and trigger unexpected consequences in L1 (like for example flap auctions or any for of kiln sell out that comes from DAI profit).

Reply:
Hmmmm yeah it would be a different approach, as using `vat.fold` we would need to adapt the oracle feed to always keep updating the price according to the new relationship (`ink`/`debt`). We wouldn't be updating `ink` anymore. That might also require some calculation changes at the Global Shutdown moment (claim token).
The way is done now is very similar to how we solve the vaults backing D3Ms.

I'd believe it deserves further investigation but if happens might be better to discuss and develop it in a separate PR.

Reply:
Resolving for now and keeping a place for future discussion here: https://github.com/makerdao/dss-bridge/issues/17